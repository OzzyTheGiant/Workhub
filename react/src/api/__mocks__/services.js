const services = { // all services must successHandler(an object literal that will update the state
	login:jest.fn().mockImplementation(({data, successHandler, errorHandler}) => {
		if (data.username === "OzzyTheGiant" && data.password === "1234567") {
			successHandler({
				isLoggedIn:true,
				currentModule:{
					name:"documents",
					listType:"clients",
					filePath:null,
					textFileData:null,
					parentId:null
				}
			});
		} else {
			errorHandler({response:{status:401}});
		}
	}),

    logout:jest.fn().mockImplementation(({successHandler, errorHandler}) => successHandler({isLoggedIn:false})),

    getClients:jest.fn().mockImplementation(({successHandler, errorHandler}) => {
		successHandler({
			clients:{
				"100000":{id:"100000", clientName: "Ozzy Perez"},
				"100001":{id:"100001", clientName: "Alondra Perez"}
			}
		});
	}),

    getProjects:jest.fn().mockImplementation(({successHandler, errorHandler}) => {
		successHandler({
			projects: {
				1:{
					category:{description:"Tax Return", id:1},
					dateCreated:"2018-04-26T19:58:20.000+0000",
					dateDue:"2017-04-15",
					id:1,
					name:"2017 Form 1040",
					client:"100000"
				}
			}
		});
	}),
	
	getDocuments:jest.fn().mockImplementation(({successHandler, errorHandler}) => {
		successHandler({
			documents:{
				"AAAAAAAAAAA":{
					accessLevel:"PUBLIC",
					category:{description:"Workpapers", id:3},
					description:"file2",
					fileType:"PDF",
					id:"AAAAAAAAAAA",
					year:2017,
					project:1
				},
				"ASDFHOWEOUH":{
					accessLevel:"PUBLIC",
					category:{description:"Notes", id:4},
					description:"notes",
					fileType:"TXT",
					id:"ASDFHOWEOUH",
					year:2017,
					project:1,
					client:"100000"
				},
				"BBBBBBBBBBB":{
					accessLevel:"PUBLIC",
					category:{description:"Workpapers", id:3},
					description:"file3",
					fileType:"EXCEL",
					id:"BBBBBBBBBBB",
					year:2017,
					project:1,
					client:"100000"
				}
			}
		});
    }),
    
	openDocument:jest.fn().mockImplementation(({data:docID, successHandler, errorHandler}) => {
		let filePath;
		switch(docID) {
			case "AAAAAAAAAAA": successHandler("/[100000] Ozzy Perez/[Tax Return] 2017 Form 1040/[AAAAAAAAAAA][Work Papers][2017] file2.pdf"); break;
			case "ASDFHOWEOUH": 
				filePath = "/[100000] Ozzy Perez/[Tax Return] 2017 Form 1040/[ASDFHOWEOUH][Notes][2017] notes.txt";
				services.downloadFile({data:filePath, successHandler, errorHandler});
				successHandler((prevState, props) => {
					return { currentModule:{...prevState.currentModule, filePath} };
				}); break;
			case "BBBBBBBBBBB":
				filePath = "/[100000] Ozzy Perez/[Tax Return] 2017 Form 1040/[BBBBBBBBBBB][Work Papers][2017] file3.xlsx"
				services.downloadFile({data:filePath, successHandler, errorHandler});
				successHandler((prevState, props) => {
					return { currentModule:{...prevState.currentModule, filePath} };
				}); break;
			default: successHandler(null); break;
		}
	}),

    downloadFile:jest.fn().mockImplementation(({data:filePath, successHandler, errorHandler}) => { 
		if (filePath && filePath.split(".").pop() === "txt") successHandler((prevState, props) => {
			return { currentModule:{...prevState.currentModule, textFileData:"Testing!"} };
		});
	}),

	getDocumentHistory:jest.fn().mockImplementation(({data:docId, successHandler, errorHandler}) => {
		successHandler({
			documentHistory:[
				{
					actionType:"OPEN",
					id:1,
					employee:{
						firstName:"John",
						middleName:null,
						lastName:"Doe",
						role:"STAFF",
						id:2
					}
				}
			]
		});
	})
};

export default services;