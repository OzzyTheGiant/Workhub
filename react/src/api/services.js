import ajax from 'api/ajax-manager';
import qs from 'qs';
import downloadDocument from 'utilities/document-downloader';

const services = { // all services must return an object literal that will update the state
	login:({data, successHandler, errorHandler}) => ajax.post("login", qs.stringify(data), {
		withCredentials:false,
		headers: {'Content-Type':'application/x-www-form-urlencoded'}
	}).then(response => successHandler({
		isLoggedIn:true,
		currentModule:{
			name:"documents",
			listType:"clients",
			filePath:null,
			textFileData:null,
			parentId:null
		}
	})).catch(error => errorHandler(error)),

    logout:({successHandler, errorHandler}) => ajax.get("logout", {
        withCredentials:false
    }).then(response => successHandler({isLoggedIn:false})).catch(error => errorHandler(error)),

    getClients:({successHandler, errorHandler}) => ajax.get("clients")
    .then(response => { 
		successHandler({clients:response.data.reduce(arrayToObjectConverter, {})});
	}).catch(error => errorHandler(error)),

    getProjects:({successHandler, errorHandler}) => ajax.get("projects")
    .then(response => { 
		successHandler({projects:response.data.reduce(arrayToObjectConverter, {})});
	}).catch(error => errorHandler(error)),
	
	getDocuments:({successHandler, errorHandler}) => ajax.get("documents")
    .then(response => { 
		successHandler({documents:response.data.reduce(arrayToObjectConverter, {})});
	}).catch(error => errorHandler(error)),
	
    // getProjectsByClientID:(clientID, successHandler, errorHandler) => ajax.get(`clients/${clientID}/projects`)
    // .then(response => successHandler(response.data)).catch(error => errorHandler(error)),

    // getDocumentsByProjectID:(projectID, successHandler, errorHandler) => ajax.get(`projects/${projectID}/documents`)
    // .then(response => successHandler(response.data)).catch(error => errorHandler(error)),
    
    openDocument:({data:docID, successHandler, errorHandler}) => ajax.get(`documents/${docID}/open`, {responseType:'text'}) // data is document ID
    .then(response => {
		let filePath = response.data;
		if (filePath.split(".").pop() !== ("pdf")) services.downloadFile({data:filePath, successHandler, errorHandler});
		else successHandler((prevState, props) => {
			return { currentModule:{...prevState.currentModule, filePath} };
		});
	}).catch(error => errorHandler(error)),

    downloadFile:({data:filePath, successHandler, errorHandler}) => ajax.get(`document-files${filePath}`, {
		responseType:filePath.split(".").pop() === "txt" ? 'text' : 'blob'
	}).then(response => {
        let url = response.config.url.split("/");
        if (response.config.responseType === 'text') {
			successHandler((prevState, props) => {
				return { currentModule:{...prevState.currentModule, textFileData:response.data} };
			});
		} else {
			successHandler((prevState, props) => {
				return { currentModule:{...prevState.currentModule, filePath} };
			});
			downloadDocument(url[url.length - 1], response.data, response.headers["content-type"])
		};
    }).catch(error => errorHandler(error)),

    getDocumentHistory:({data:docID, successHandler, errorHandler}) => ajax.get(`documents/${docID}/history`)
    .then(response => successHandler({documentHistory:response.data})).catch(error => errorHandler(error))
};

function arrayToObjectConverter (accumulator, entity, i, entities) {
	/* convert from array to object with id keys */
	accumulator[entity.id] = entities[i];
	return accumulator;
}

export default services;

