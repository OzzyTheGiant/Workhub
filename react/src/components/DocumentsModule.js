import React from 'react';
import PropTypes from 'prop-types';
import Module from 'components/Module';

class DocumentsModule extends React.Component  {
    constructor() {
        super();
        this.state = {displayType:"grid-view"}
    }

    createList = (clientID, projectID) => {
        let data = this.props.clients;
        // let projects;
        if (clientID) { // get list of projects
            data = data.filter(client => client.id === clientID)[0];
            //if (!data.projects) projects = this.props.getProjects(clientID);
            if (projectID) { // get list of documents
                data = data.filter(project => project.id === projectID);
            }
        }
        this.setState(data)
    }

    toggleDisplayType = () => {
        this.setState({
            dipslayType:this.state.displayType === "grid-view" ? "list-view" : "grid-view"
        });
    }

	render() {
        const clients = this.props.clients;
		return (
            <Module title="Documents">
                <ul className={this.state.displayType}>
                { clients.map((client, index) => <li key={client.id}>{client.clientName}</li>) }
                </ul>
            </Module>
        );
	}
}

DocumentsModule.propTypes = {
	clients:PropTypes.array,
}

export default DocumentsModule;