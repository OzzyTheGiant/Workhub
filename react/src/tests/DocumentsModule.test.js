import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import DocumentsModule from 'components/DocumentsModule';
jest.mock('api/services');
import services from 'api/services';

describe("DocumentsModule", () => {
	// appState represents state in App component, which is a placeholder for props to rerender the component
	let componentWrapper = null, documentsModule = null, appState = {}, errorHandler = jest.fn();
	/* ==== MOCKS ==== */
	const updateGlobalState = jest.fn().mockImplementation((props) => {
		/* update app's state, stored in appState variable */
		if (typeof props === "function") {
			appState = {...appState, ...props(appState)};
		} else {
			appState = {...appState, ...props};
		}
	});
	const serviceCaller = jest.fn().mockImplementation(function(service, postData) {
		/* when testing componentDidMount(), instead of calling App's method, it calls this function
			* which will return fake data from mock services module. Instead of setting props directly on currentWrapper,
			* due to circular reference issues and componentWrapper being null at the time of creating the successHandler function,
			* a brand new component and wrapper will have to be instantiated but with previous component's props */
		switch(service) {
			case "getClients": services.getClients({
				successHandler:data => appState.clients = data.clients,
				errorHandler
			}); break;
			case "getProjects": services.getProjects({
				successHandler:data => appState.projects = data.projects,
				errorHandler
			}); break;
			case "getDocuments": services.getDocuments({
				successHandler:data => appState.documents = data.documents,
				errorHandler
			}); break;
			case "openDocument": services.openDocument({
				data:postData,
				successHandler:updateGlobalState,
				errorHandler
			}); break;
			case "downloadFile": services.downloadFile({
				data:postData,
				successHandler:updateGlobalState,
				errorHandler
			}); break;
			case "getDocumentHistory": services.getDocumentHistory({
				data:postData,
				successHandler:data => appState.documentHistory = data.documentHistory,
				errorHandler
			})
		} 
	});

    /* ==== COMPONENT ==== */
	let initComponent = () => {
		return (
			<DocumentsModule 
			clients={null}
			projects={null}
			documents={null}
			documentHistory={[]}
			breadCrumbs={[]}
			clickSelection={null}
			iconDisplayType={"grid-view"}
			currentModule={{
				name:"documents",
				listType:"clients",
				filePath:null,
				textFileData:null,
				parentId:null
			}}
			serviceCaller={serviceCaller}
			updateGlobalState={updateGlobalState}
			updateModuleState={jest.fn()}/> // updateModuleState is never used, so for now we will mock it
		)
	};

	beforeAll(() => {
		configure({adapter:new Adapter()});
    });
    
    beforeEach(() => {
		documentsModule = initComponent();
		appState = {...documentsModule.props};
		componentWrapper = mount(documentsModule);
		if (appState.clients) componentWrapper.setProps({clients:appState.clients});
		if (appState.projects) componentWrapper.setProps({projects:appState.projects});
		if (appState.documents) componentWrapper.setProps({documents:appState.documents});
	});
	
	afterEach(() => {
		appState.clients = appState.projects = appState.documents = componentWrapper = documentsModule =  null;
	});

	it("Should render folder lists correctly", () => {
		expect(toJson(componentWrapper)).toMatchSnapshot();
    });

	/* instance() calls used to be "doubleClick" simulators but broke because event.currentTarget doesn't work
	 * which is then followed by .setProps() to update the component */
    it("Should render file lists correctly", () => {
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"1"}}});
        componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
        expect(toJson(componentWrapper)).toMatchSnapshot();
    })
    
    it("Should change icon view and show/hide details when clicking toggleIconView button", () => {
		expect(componentWrapper.find("ul").prop("className")).toBe("frame grid-view");
		componentWrapper.find("button").at(0).simulate('click');
		componentWrapper.setProps(appState);
		expect(componentWrapper.find("ul").prop("className")).toBe("frame list-view");
        expect(componentWrapper.find(".details p").at(0).text()).toBe("100001");
		componentWrapper.find("button").at(0).simulate('click');
		componentWrapper.setProps(appState);
        expect(componentWrapper.find("ul").prop("className")).toBe("frame grid-view");
        expect(componentWrapper.find(".details").children().length).toBe(0);
    });
	
    it("Should render projects after double-clicking a client", () => {
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(1).simulate("doubleClick");
		componentWrapper.setProps(appState);
		let currentModule = componentWrapper.prop("currentModule");
		expect(currentModule.listType).toBe("projects")
        expect(currentModule.parentId).toBe("100000");
        expect(componentWrapper.find("li").at(0).prop("title")).toBe("2017 Form 1040");
    });

    it("Should render documents after double-clicking a project", () => {
        componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"1"}}});
        componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
		expect(appState.currentModule.listType).toBe("documents");
        expect(componentWrapper.prop("currentModule").parentId).toBe("1");
        expect(componentWrapper.find("li").at(0).prop("title")).toBe("file2");
    });

    it("Should render breadcrumbs correctly after double-clicking an icon", () => {
        expect(componentWrapper.find("#breadcrumb-slider").children().length).toBe(0);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(1).simulate("doubleClick");
        componentWrapper.setProps(appState);
        expect(componentWrapper.find("#breadcrumb-slider").children().at(0).text()).toBe("Home");
        expect(componentWrapper.find("#breadcrumb-slider").children().at(1).text()).toBe("Ozzy Perez");
        componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:1}}});
		componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
        expect(componentWrapper.find("#breadcrumb-slider").children().at(2).text()).toBe("2017 Form 1040");
    });

    it ("Should render download button if file is not PDF and call downloadFile() when button is clicked", () => {
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(1).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"1"}}});
        componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"BBBBBBBBBBB"}}});
		componentWrapper.setProps(appState);
		componentWrapper.find(".grid-view li").at(2).simulate("doubleClick");
		componentWrapper.setProps(appState)
		expect(services.openDocument).toBeCalled();
		componentWrapper.find("#download-prompt .color-button").at(0).simulate("click");
		expect(services.downloadFile).toHaveBeenCalledTimes(2);
        expect(services.downloadFile).toBeCalledWith({
			data:"/[100000] Ozzy Perez/[Tax Return] 2017 Form 1040/[BBBBBBBBBBB][Work Papers][2017] file3.xlsx",
			errorHandler,
			successHandler:updateGlobalState
		});
    });

    it("Should display text from a text file once it has been received", () => {
        componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(1).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"1"}}});
        componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"ASDFHOWEOUH"}}});
		componentWrapper.setProps(appState);
		componentWrapper.find(".grid-view li").at(1).simulate("doubleClick");
		componentWrapper.setProps(appState);
        expect(componentWrapper.find("#text-file").text()).toBe("Testing!");
    });

    it("Should go to previous level in list hierarchy when clicking on a breadcrumb", () => {
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100001"}}});
		componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
        componentWrapper.setProps(appState);
        componentWrapper.instance().navigationHandler({target:{dataset:{level:"1"}, parentNode:{id:"breadcrumb-slider"}, tagName:"DIV"}}, )
		componentWrapper.setProps(appState);
		expect(appState.currentModule.listType).toBe("clients");
		expect(componentWrapper.find("li").at(0).text()).toBe("Alondra Perez");
    });

    it("Should render document history when clicking toolbar button", () => {
        componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"100000"}}});
		componentWrapper.find(".grid-view li").at(1).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"1"}}});
        componentWrapper.find(".grid-view li").at(0).simulate("doubleClick");
		componentWrapper.setProps(appState);
		componentWrapper.instance().iconClickHandler({currentTarget:{dataset:{id:"ASDFHOWEOUH"}}});
		componentWrapper.setProps(appState);
		componentWrapper.find("button").at(1).simulate("click");
		componentWrapper.setProps(appState);
		expect(services.getDocumentHistory).toBeCalled();
        expect(componentWrapper.find("#DocumentHistory").type()).toBe("ol");
        expect(componentWrapper.find("#DocumentHistory > li").length).toBe(2);
    });
});