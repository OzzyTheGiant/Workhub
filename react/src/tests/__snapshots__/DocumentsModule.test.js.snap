// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsModule Should render file lists correctly 1`] = `
<DocumentsModule
  breadCrumbs={
    Array [
      Object {
        "clickHandler": [Function],
        "text": "Home",
      },
      Object {
        "clickHandler": [Function],
        "text": "Ozzy Perez",
      },
      Object {
        "clickHandler": [Function],
        "text": "2017 Form 1040",
      },
    ]
  }
  clickSelection={null}
  clients={
    Object {
      "100000": Object {
        "clientName": "Ozzy Perez",
        "id": "100000",
      },
      "100001": Object {
        "clientName": "Alondra Perez",
        "id": "100001",
      },
    }
  }
  currentModule={
    Object {
      "filePath": null,
      "listType": "documents",
      "name": "documents",
      "parentId": "1",
      "textFileData": null,
    }
  }
  documentHistory={Array []}
  documents={
    Object {
      "AAAAAAAAAAA": Object {
        "accessLevel": "PUBLIC",
        "category": Object {
          "description": "Workpapers",
          "id": 3,
        },
        "description": "file2",
        "fileType": "PDF",
        "id": "AAAAAAAAAAA",
        "project": 1,
        "year": 2017,
      },
      "ASDFHOWEOUH": Object {
        "accessLevel": "PUBLIC",
        "category": Object {
          "description": "Notes",
          "id": 4,
        },
        "client": "100000",
        "description": "notes",
        "fileType": "TXT",
        "id": "ASDFHOWEOUH",
        "project": 1,
        "year": 2017,
      },
      "BBBBBBBBBBB": Object {
        "accessLevel": "PUBLIC",
        "category": Object {
          "description": "Workpapers",
          "id": 3,
        },
        "client": "100000",
        "description": "file3",
        "fileType": "EXCEL",
        "id": "BBBBBBBBBBB",
        "project": 1,
        "year": 2017,
      },
    }
  }
  iconDisplayType="grid-view"
  projects={
    Object {
      "1": Object {
        "category": Object {
          "description": "Tax Return",
          "id": 1,
        },
        "client": "100000",
        "dateCreated": "2018-04-26T19:58:20.000+0000",
        "dateDue": "2017-04-15",
        "id": 1,
        "name": "2017 Form 1040",
      },
    }
  }
  serviceCaller={[Function]}
  updateGlobalState={[Function]}
  updateModuleState={[Function]}
>
  <Module
    buttonActions={
      Array [
        [Function],
        null,
      ]
    }
    iconDisplayType="grid-view"
    title="Documents"
  >
    <div
      className="Module"
      id="DocumentsModule"
    >
      <div
        id="toolbar"
      >
        <h1>
          Documents
        </h1>
        <div
          className="toolbarButtons"
        >
          <button
            className="toggleIconView toolbar-button button-grid-view"
            key="0"
            onClick={[Function]}
          />
        </div>
      </div>
      <Breadcrumbs
        breadcrumbs={
          Array [
            Object {
              "clickHandler": [Function],
              "text": "Home",
            },
            Object {
              "clickHandler": [Function],
              "text": "Ozzy Perez",
            },
            Object {
              "clickHandler": [Function],
              "text": "2017 Form 1040",
            },
          ]
        }
      >
        <div
          id="breadcrumbs"
        >
          <div
            id="breadcrumb-slider"
          >
            <div
              data-level={1}
              key="0"
              onClick={[Function]}
            >
              Home
            </div>
            <div
              data-level={2}
              key="1"
              onClick={[Function]}
            >
              Ozzy Perez
            </div>
            <div
              data-level={3}
              key="2"
              onClick={[Function]}
            >
              2017 Form 1040
            </div>
          </div>
        </div>
      </Breadcrumbs>
      <IconLinkList
        clickSelection={null}
        displayType="grid-view"
        iconClickHandler={[Function]}
        iconDoubleClickHandler={[Function]}
        list={
          Object {
            "AAAAAAAAAAA": Object {
              "accessLevel": "PUBLIC",
              "category": Object {
                "description": "Workpapers",
                "id": 3,
              },
              "description": "file2",
              "fileType": "PDF",
              "id": "AAAAAAAAAAA",
              "project": 1,
              "year": 2017,
            },
            "ASDFHOWEOUH": Object {
              "accessLevel": "PUBLIC",
              "category": Object {
                "description": "Notes",
                "id": 4,
              },
              "client": "100000",
              "description": "notes",
              "fileType": "TXT",
              "id": "ASDFHOWEOUH",
              "project": 1,
              "year": 2017,
            },
            "BBBBBBBBBBB": Object {
              "accessLevel": "PUBLIC",
              "category": Object {
                "description": "Workpapers",
                "id": 3,
              },
              "client": "100000",
              "description": "file3",
              "fileType": "EXCEL",
              "id": "BBBBBBBBBBB",
              "project": 1,
              "year": 2017,
            },
          }
        }
        nameLabel="description"
        type="documents"
      >
        <ul
          className="frame grid-view"
        >
          <IconLink
            dblClickHandler={[Function]}
            details={
              Object {
                "category": "Workpapers",
                "year": 2017,
              }
            }
            fileType="PDF"
            iconClickHandler={[Function]}
            id="AAAAAAAAAAA"
            key="AAAAAAAAAAA"
            name="file2"
            selected={false}
            view="grid-view"
          >
            <li
              className=" PDF"
              data-id="AAAAAAAAAAA"
              onClick={[Function]}
              onDoubleClick={[Function]}
              title="file2"
            >
              <p>
                file2
              </p>
              <span
                className="details"
              />
            </li>
          </IconLink>
          <IconLink
            dblClickHandler={[Function]}
            details={
              Object {
                "category": "Workpapers",
                "year": 2017,
              }
            }
            fileType="EXCEL"
            iconClickHandler={[Function]}
            id="BBBBBBBBBBB"
            key="BBBBBBBBBBB"
            name="file3"
            selected={false}
            view="grid-view"
          >
            <li
              className=" EXCEL"
              data-id="BBBBBBBBBBB"
              onClick={[Function]}
              onDoubleClick={[Function]}
              title="file3"
            >
              <p>
                file3
              </p>
              <span
                className="details"
              />
            </li>
          </IconLink>
          <IconLink
            dblClickHandler={[Function]}
            details={
              Object {
                "category": "Notes",
                "year": 2017,
              }
            }
            fileType="TXT"
            iconClickHandler={[Function]}
            id="ASDFHOWEOUH"
            key="ASDFHOWEOUH"
            name="notes"
            selected={false}
            view="grid-view"
          >
            <li
              className=" TXT"
              data-id="ASDFHOWEOUH"
              onClick={[Function]}
              onDoubleClick={[Function]}
              title="notes"
            >
              <p>
                notes
              </p>
              <span
                className="details"
              />
            </li>
          </IconLink>
        </ul>
      </IconLinkList>
    </div>
  </Module>
</DocumentsModule>
`;

exports[`DocumentsModule Should render folder lists correctly 1`] = `
<DocumentsModule
  breadCrumbs={Array []}
  clickSelection={null}
  clients={
    Object {
      "100000": Object {
        "clientName": "Ozzy Perez",
        "id": "100000",
      },
      "100001": Object {
        "clientName": "Alondra Perez",
        "id": "100001",
      },
    }
  }
  currentModule={
    Object {
      "filePath": null,
      "listType": "clients",
      "name": "documents",
      "parentId": null,
      "textFileData": null,
    }
  }
  documentHistory={Array []}
  documents={
    Object {
      "AAAAAAAAAAA": Object {
        "accessLevel": "PUBLIC",
        "category": Object {
          "description": "Workpapers",
          "id": 3,
        },
        "description": "file2",
        "fileType": "PDF",
        "id": "AAAAAAAAAAA",
        "project": 1,
        "year": 2017,
      },
      "ASDFHOWEOUH": Object {
        "accessLevel": "PUBLIC",
        "category": Object {
          "description": "Notes",
          "id": 4,
        },
        "client": "100000",
        "description": "notes",
        "fileType": "TXT",
        "id": "ASDFHOWEOUH",
        "project": 1,
        "year": 2017,
      },
      "BBBBBBBBBBB": Object {
        "accessLevel": "PUBLIC",
        "category": Object {
          "description": "Workpapers",
          "id": 3,
        },
        "client": "100000",
        "description": "file3",
        "fileType": "EXCEL",
        "id": "BBBBBBBBBBB",
        "project": 1,
        "year": 2017,
      },
    }
  }
  iconDisplayType="grid-view"
  projects={
    Object {
      "1": Object {
        "category": Object {
          "description": "Tax Return",
          "id": 1,
        },
        "client": "100000",
        "dateCreated": "2018-04-26T19:58:20.000+0000",
        "dateDue": "2017-04-15",
        "id": 1,
        "name": "2017 Form 1040",
      },
    }
  }
  serviceCaller={[Function]}
  updateGlobalState={[Function]}
  updateModuleState={[Function]}
>
  <Module
    buttonActions={
      Array [
        [Function],
        null,
      ]
    }
    iconDisplayType="grid-view"
    title="Documents"
  >
    <div
      className="Module"
      id="DocumentsModule"
    >
      <div
        id="toolbar"
      >
        <h1>
          Documents
        </h1>
        <div
          className="toolbarButtons"
        >
          <button
            className="toggleIconView toolbar-button button-grid-view"
            key="0"
            onClick={[Function]}
          />
        </div>
      </div>
      <Breadcrumbs
        breadcrumbs={Array []}
      >
        <div
          id="breadcrumbs"
        >
          <div
            id="breadcrumb-slider"
          />
        </div>
      </Breadcrumbs>
      <IconLinkList
        clickSelection={null}
        displayType="grid-view"
        iconClickHandler={[Function]}
        iconDoubleClickHandler={[Function]}
        list={
          Object {
            "100000": Object {
              "clientName": "Ozzy Perez",
              "id": "100000",
            },
            "100001": Object {
              "clientName": "Alondra Perez",
              "id": "100001",
            },
          }
        }
        nameLabel="clientName"
        type="clients"
      >
        <ul
          className="frame grid-view"
        >
          <IconLink
            dblClickHandler={[Function]}
            details={
              Object {
                "id": "100001",
              }
            }
            fileType={null}
            iconClickHandler={[Function]}
            id="100001"
            key="100001"
            name="Alondra Perez"
            selected={false}
            view="grid-view"
          >
            <li
              className=" null"
              data-id="100001"
              onClick={[Function]}
              onDoubleClick={[Function]}
              title="Alondra Perez"
            >
              <p>
                Alondra Perez
              </p>
              <span
                className="details"
              />
            </li>
          </IconLink>
          <IconLink
            dblClickHandler={[Function]}
            details={
              Object {
                "id": "100000",
              }
            }
            fileType={null}
            iconClickHandler={[Function]}
            id="100000"
            key="100000"
            name="Ozzy Perez"
            selected={false}
            view="grid-view"
          >
            <li
              className=" null"
              data-id="100000"
              onClick={[Function]}
              onDoubleClick={[Function]}
              title="Ozzy Perez"
            >
              <p>
                Ozzy Perez
              </p>
              <span
                className="details"
              />
            </li>
          </IconLink>
        </ul>
      </IconLinkList>
    </div>
  </Module>
</DocumentsModule>
`;
